avocado/core/nrunner.py

```diff
@@ -100,11 +100,11 @@ class ConfigEncoder(json.JSONEncoder):
     JSON Encoder for config options.
     """
 
-    def default(self, config_option):  # pylint: disable=W0221
-        if isinstance(config_option, set):
-            return {'__encoded_set__': list(config_option)}
+    def default(self, o):
```

Comment:
Such a nice variable name in cPython... sigh. But your change looks good, thanks!